go/consensus/tendermint/apps/registry/transactions.go

```diff
@@ -241,13 +241,74 @@ func (app *registryApplication) registerNode( // nolint: gocyclo
 
 	// Check runtime's whitelist.
 	for _, rt := range paidRuntimes {
-		if rt.AdmissionPolicy.EntityWhitelist != nil && !rt.AdmissionPolicy.EntityWhitelist.Entities[newNode.EntityID] {
+		if rt.AdmissionPolicy.EntityWhitelist == nil {
+			continue
+		}
+		wcfg, entIsWhitelisted := rt.AdmissionPolicy.EntityWhitelist.Entities[newNode.EntityID]
+		if !entIsWhitelisted {
 			ctx.Logger().Error("RegisterNode: node's entity not in a runtime's whitelist",
 				"entity", newNode.EntityID,
 				"runtime", rt.ID,
 			)
 			return registry.ErrForbidden
 		}
+		if wcfg.MaxNodes == nil {
+			continue
+		}
+		if len(wcfg.MaxNodes) == 0 {
+			continue
+		}
+
+		// Map is present and non-empty, check per-role restrictions
+		// on the maximum number of nodes per entity.
+
```

Comment:
The below code needs some test coverage.

Reply:
Yeah, I'm adding that now :)